From 523e09eb203b68b9135341dbda7fe353aa6987d3 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 16 Nov 2023 07:06:27 +0100
Subject: [PATCH 0070/1695] dts: stm32mp15: harden RCC secure configuration on
 ST boards

Enable STM32MP15 RCC secure hardening configuration on ST boards
(DK1, DK2, ED1 and EV1) to assign SoC clocks, reset controllers
and PWR regulators to OP-TEE secure world.

This change removes setting of &rcc node status property from
stm32mp157a-dk1.dts, stm32mp157c-dk2.dts as the property is
set from stm32mp15xx-dkx.dtsi that is included from the 2 former
DTS files.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c419aa5369e05671c8b5363a3a6465a646938dd3 dts: stm32mp15: harden RCC secure configuration on ST boards
00324efc268d2b3e88b030c049aca03fe0688d58 dts: stm32mp15: harden RCC secure configuration on ST boards]
Change-Id: I8f615a9c9b28a4fa5b64fbc251a24f8ce68e5ecb
[etienne: cherry-picked from 08278885852f32adf58750d0241b543218249f95]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/dts/stm32mp157a-dk1.dts  | 4 ----
 core/arch/arm/dts/stm32mp157c-dk2.dts  | 4 ----
 core/arch/arm/dts/stm32mp157c-ed1.dts  | 1 +
 core/arch/arm/dts/stm32mp15xx-dkx.dtsi | 5 +++++
 4 files changed, 6 insertions(+), 8 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp157a-dk1.dts b/core/arch/arm/dts/stm32mp157a-dk1.dts
index 7ba971e5b..294746da4 100644
--- a/core/arch/arm/dts/stm32mp157a-dk1.dts
+++ b/core/arch/arm/dts/stm32mp157a-dk1.dts
@@ -36,7 +36,3 @@
 		reg = <0xf0 0x10>;
 	};
 };
-
-&rcc {
-	status = "okay";
-};
diff --git a/core/arch/arm/dts/stm32mp157c-dk2.dts b/core/arch/arm/dts/stm32mp157c-dk2.dts
index b82730cbc..dfb8f9830 100644
--- a/core/arch/arm/dts/stm32mp157c-dk2.dts
+++ b/core/arch/arm/dts/stm32mp157c-dk2.dts
@@ -98,10 +98,6 @@
 	};
 };
 
-&rcc {
-	status = "okay";
-};
-
 &usart2 {
 	pinctrl-names = "default", "sleep", "idle";
 	pinctrl-0 = <&usart2_pins_c>;
diff --git a/core/arch/arm/dts/stm32mp157c-ed1.dts b/core/arch/arm/dts/stm32mp157c-ed1.dts
index 2edb4d42b..eb679f5db 100644
--- a/core/arch/arm/dts/stm32mp157c-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157c-ed1.dts
@@ -338,6 +338,7 @@
 };
 
 &rcc {
+	compatible = "st,stm32mp1-rcc-secure";
 	status = "okay";
 };
 
diff --git a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
index 793bf4cc9..a0d4eb20b 100644
--- a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
@@ -489,6 +489,11 @@
 	vdd_3v3_usbfs-supply = <&vdd_usb>;
 };
 
+&rcc {
+	compatible = "st,stm32mp1-rcc-secure";
+	status = "okay";
+};
+
 &rng1 {
 	status = "okay";
 };
-- 
2.47.2

